### ISSUE TITLE
Metadata marked as required but gets undefined

### ISSUE THREAD

**Issue by kirkegaard** [catamphetamine/react-phone-number-input#426]
When using `import PhoneInputWithCountry from "react-phone-number-input/react-hook-form";` im getting a warning:

<img width="514" alt="Screenshot 2023-08-07 at 14 24 29" src="https://github.com/catamphetamine/react-phone-number-input/assets/43752/80180bf3-bfb4-470d-8a4b-aa63c915c1a3">

```bash
$ yarn why react-hook-form
yarn why v1.22.19
[1/4] 🤔  Why do we have the module "react-hook-form"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "react-hook-form@7.45.4"
info Has been hoisted to "react-hook-form"
info This module exists because it's specified in "dependencies".
info Disk size without dependencies: "1.91MB"
info Disk size with unique dependencies: "1.91MB"
info Disk size with transitive dependencies: "1.91MB"
info Number of shared dependencies: 0
✨  Done in 0.17s.

$ yarn why react-phone-number-input
yarn why v1.22.19
[1/4] 🤔  Why do we have the module "react-phone-number-input"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "react-phone-number-input@3.3.1"
info Has been hoisted to "react-phone-number-input"
info This module exists because it's specified in "dependencies".
info Disk size without dependencies: "9.73MB"
info Disk size with unique dependencies: "43.2MB"
info Disk size with transitive dependencies: "43.32MB"
info Number of shared dependencies: 9
✨  Done in 0.35s.

$ yarn why libphonenumber-js
yarn why v1.22.19
[1/4] 🤔  Why do we have the module "libphonenumber-js"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "libphonenumber-js@1.10.39"
info Reasons this module exists
   - "react-phone-number-input" depends on it
   - Hoisted from "react-phone-number-input#libphonenumber-js"
info Disk size without dependencies: "9.84MB"
info Disk size with unique dependencies: "9.84MB"
info Disk size with transitive dependencies: "9.84MB"
info Number of shared dependencies: 0
✨  Done in 0.18s.
```

**Comment by kirkegaard** [comment]
https://codesandbox.io/p/sandbox/jolly-shape-9np7rm

**Comment by Prooksius** [comment]
The same problem.
Temporarily fixed this by importing metadata and explicitly assigning it to PhoneInput.

```
import metadata from 'libphonenumber-js/min/metadata'
...
<PhoneInputWithCountry
	international
	metadata={metadata}

```

**Comment by Haraldson** [comment]
Experiencing the same as well.

**Comment by catamphetamine** [comment]
Turns out that the previous refactoring that removed `.defaultProps` has left `.isRequired` on those props and those `.isRequired` statements should've been removed too.

Published `react-phone-number-input@3.3.2`.
